Sources/Basic/FileSystem.swift

```diff
@@ -118,6 +118,11 @@ public protocol FileSystem {
     //
     // FIXME: This is obviously not a very efficient or flexible API.
     mutating func writeFileContents(_ path: AbsolutePath, bytes: ByteString) throws
+
+    /// Open an output stream to the given file path.
+    ///
+    /// - Returns: stream to the given file path.
+    func openFileStream(_ path: AbsolutePath) throws -> FileOutputByteStream
```

Comment:
I don't know what to call this, but it should probably clearly indicate this is a *writeable* stream... we may one day have the same concept for input streams.